src/app/leaves/actions/updated-leave-request.ts

```diff
@@ -0,0 +1,29 @@
+"use server";
+
+import { Tables } from "@/lib/db";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+import { leaveRequestSchema } from "../add/components/LeaveRequestFormBuilder";
+import { z } from "zod";
+import { LeaveRequestStatus } from "@/constant/constant";
+
+export const updateLeaveRequest = async (
+  requestId: number,
+  values: z.infer<typeof leaveRequestSchema>
+) => {
+  const supabase = await getSupabaseClient();
+  const { error } = await supabase
+    .from(Tables.Leaves)
```

Comment:
We need to add a check here. Leave request can only be updated if it's not approved yet.

Reply:
Sir @Monfernape I think we don't need to add a check because the payload contains a static value of `Pending`, and we are not passing this value as a prop.

Reply:
Right but this method can be called when user is trying to update already approved leave. That's why we need to make sure we block this flow.